src/background/injector-controller.ts

```diff
@@ -42,22 +39,18 @@ export class InjectorController {
             inspectStoreInjectingRequested || visualizationStoreState.injectingRequested;
 
         if (isInjectingRequested && !visualizationStoreState.injectingStarted) {
-            this.setTimeout(() => {
-                this.interpreter.interpret({
-                    messageType: Messages.Visualizations.State.InjectionStarted,
-                    tabId: tabId,
-                });
-            }, InjectorController.injectionStartedWaitTime);
+            this.interpreter.interpret({
+                messageType: Messages.Visualizations.State.InjectionStarted,
+                tabId: tabId,
+            });
 
             this.injector
                 .injectScripts(tabId)
                 .then(async () => {
-                    const response = this.interpreter.interpret({
+                    await this.interpreter.interpret({
                         messageType: Messages.Visualizations.State.InjectionCompleted,
                         tabId: tabId,
-                    });
-
-                    await response.result;
+                    }).result;
                 })
                 .catch(this.logger.error);
```

Comment:
I don't think .then() and .catch() actually wait for the promise in question, could this be a try/catch block instead?